contracts/0.8.9/lib/PositiveTokenRebaseLimiter.sol

```diff
@@ -65,21 +99,27 @@ library PositiveTokenRebaseLimiter {
      * @return true if limit is reached
      */
     function isLimitReached(TokenRebaseLimiterData memory _limiterState) internal pure returns (bool) {
-        return _limiterState.accumulatedRebase == _limiterState.rebaseLimit;
+        if (_limiterState.rebaseLimit == UNLIMITED_REBASE) return false;
+        if (_limiterState.postTotalPooledEther < _limiterState.preTotalPooledEther) return false;
+
+        uint256 accumulatedEther = _limiterState.postTotalPooledEther - _limiterState.preTotalPooledEther;
+        uint256 accumulatedRebase;
+
+        if (_limiterState.preTotalPooledEther > 0) {
+            accumulatedRebase = accumulatedEther * LIMITER_PRECISION_BASE / _limiterState.preTotalPooledEther;
+        }
+
+        return accumulatedRebase >= _limiterState.rebaseLimit;
     }
 
     /**
      * @notice raise limit using the given amount of ether
      * @param _limiterState limit repr struct
      */
     function raiseLimit(TokenRebaseLimiterData memory _limiterState, uint256 _etherAmount) internal pure {
-        if(_limiterState.rebaseLimit == UNLIMITED_REBASE) { return; }
-
-        uint256 projectedLimit = _limiterState.rebaseLimit + (
-            _etherAmount * LIMITER_PRECISION_BASE
-        ) / _limiterState.totalPooledEther;
+        if (_limiterState.rebaseLimit == UNLIMITED_REBASE) return;
 
-        _limiterState.rebaseLimit = Math256.min(projectedLimit, UNLIMITED_REBASE);
+        _limiterState.postTotalPooledEther -= _etherAmount;
```

Comment:
can't there be a situation where `postTotalPooledEther` < `_etherAmount` which will lead to a revert?
let's say `_preCLBalance` > 0, `_postCLBalance` = 0, `_etherToLockForWithdrawals` anything > 0

Reply:
then let's revert indeed :eyes:

Reply:
added